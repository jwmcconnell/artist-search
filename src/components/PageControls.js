import React from 'react';
import PropTypes from 'prop-types';

const styles = {
  controls: {
    textAlign: 'center',
    margin: '10px'
  },
  button: {
    margin: '5px'
  }
};

const PageControls = ({ currentPage, pages, handlePageChange, disableControls }) => {
  return (
    <section style={styles.controls}>
      <button 
        style={styles.button} 
        onClick={() => handlePageChange('decrement')}
        disabled={currentPage <= 1 || disableControls}
      >←</button>
      <span>Page: {currentPage} of: {pages}</span>
      <button 
        style={styles.button} 
        onClick={() => handlePageChange('increment')}
        disabled={currentPage === pages || disableControls}
      >→</button>
    </section>
  );
};

PageControls.propTypes = {
  currentPage: PropTypes.number.isRequired,
  pages: PropTypes.number.isRequired,
  handlePageChange: PropTypes.func.isRequired,
  disableControls: PropTypes.bool.isRequired
};

export default PageControls; 
