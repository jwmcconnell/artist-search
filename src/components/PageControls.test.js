import React from 'react';
import { shallow } from 'enzyme';
import PageControls from './PageControls';

describe('PageControls component', () => {
  it('renders PageControls', () => {
    const wrapper = shallow(<PageControls 
      currentPage={1}
      pages={10} 
      handlePageChange={() => {}} 
      disableControls={false}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
