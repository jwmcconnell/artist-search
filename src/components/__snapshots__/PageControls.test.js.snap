// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageControls component renders PageControls 1`] = `
<section
  style={
    Object {
      "margin": "10px",
      "textAlign": "center",
    }
  }
>
  <button
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "margin": "5px",
      }
    }
  >
    ←
  </button>
  <span>
    Page: 
    1
     of: 
    10
  </span>
  <button
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "margin": "5px",
      }
    }
  >
    →
  </button>
</section>
`;
